### PR TITLE
Save segment and stem properties correctly

### PR DESCRIPTION
**Pull request by peterhieuvu**
Two saving issues fixed here. The first one was leading spaces which are saved only when the element is no generated. This is normally fine, but segments are a bit disconnected from the elements related to it and so changing a segment property (which includes leading space) doesn't set the generated flag properly for related elements. Since segments contain elements for different voices, I opted to just set all of the elements related to a segment to not be generated because they all should have the leading space applied. Reading and writing segments goes by the segment anyway so if I understand the code and tested it correctly it shouldn't create issues or redundancy anyway. There are also checks in place to ensure that segments are not read and written more than once.

The second issue has to do with saving stacking order for stems. Originally this was thought to be similar to the above issue, but it's actually not. Stems are not written as a directly connected element to segments, but instead as a part of a chord which makes sense. There it writes the stem if it isUserModified(). This is a function that is used in only a few places and is a bit old. That function checks the styled properties of the element and compares it to the property default. This is good, but there are also the standard properties that all elements have and can be changed through the inspector. There's a line to check for these, but it doesn't check for all the properties.

https://musescore.org/en/node/139846
https://musescore.org/en/node/279245

### CODE REVIEW

**Comment by MarcSabatella** [comment]
I had the same realization just recently about leading space being a bit different, but didn't look into it far enough to come up with a solution.  I guess setting everything to not generated makes sense.  Seems like it works for some otherwise generated elements already, though - how?  Maybe they were already being marked not generated as soon as you touch any property, like I believe is the case for timesigs?

File: libmscore/segment.cpp [hunk 1]
```diff
@@ -895,6 +895,8 @@ bool Segment::setProperty(Pid propertyId, const QVariant& v)
                   break;
             case Pid::LEADING_SPACE:
                   setExtraLeadingSpace(v.value<Spatium>());
+                  for (Element* e : _elist)
+                        if(e) e->setGenerated(false);
```
Comment (MarcSabatella) [inline libmscore/segment.cpp]: not crazy about the formatting.  If you do want to force this to one line, still have the space after the "if", but consider breaking it into lines as usual (and adding braces on the "for", even though not strictly necessary, as we do elsewhere).

Comment (peterhieuvu) [reply libmscore/segment.cpp]: My mistake, old habit of mine. I’m aware that that’s against the style guide, I’ll fix it when I get back to my computer :)

**Comment by peterhieuvu** [comment]
Yeah time signatures and bar lines are always not “generated” even though they are generated and not user modified. For that reason they’re always written and so the segment is always written. I’m not completely sure why the time signature is always not generated and written while the end bar line and the clef is not, but it’s currently the way it works. As an example if you create a one measure treble clef score, you get two segments: one for time signature and one for the measure rest.

**Pull request merged**